### PR TITLE
PMP : deal with uninitialized face_index map in border_halfedges

### PR DESCRIPTION
**Pull request by janetournois**
In `PMP::border_halfedges()` (and as a matter of facts `PMP::isotropic_remeshing()`,
one can provide a `face_index_map` as named parameter.

If it's not provided by the user, the internal `face_index` map (if it exists) is used.
It can happen that this internal property map exists, but is not initialized. It is for example the case for a `Polyhedron_with_items_3` for which the user does not set the face ids.

This PR
- documents the fact that the internal face index map, if it exists, should be initialized
- adds a warning (via `std::cerr`) before the face index map is used, and so before the crash happens, to help the user find the problem

See e.g. the issue #1017

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/border.h [hunk 1]
```diff
@@ -157,14 +159,26 @@ namespace Polygon_mesh_processing {
...
     //face index map given as a named parameter, or as an internal property map
     FIMap fim = choose_param(get_param(np, face_index),
                              get(CGAL::face_index, pmesh));
+
+    //make a minimal check that it's properly initialized :
+    //if the 2 first faces have the same id, we know the property map is not initialized
+    typename boost::range_iterator<const FaceRange>::type it = boost::const_begin(faces);
+    if (get(fim, *it++) == get(fim, *it))
+    {
+      std::cerr << "WARNING : the internal property map for CGAL::face_index_t" << std::endl
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/border.h]: is the property map always the internal one?

Comment (janetournois) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/border.h]: i thought yes, but no
i'll make a new commit
thanks

**Comment by janetournois** [comment]
@sloriot now it is!

**Comment by janetournois** [comment]
from my side, it's ready for testing

**Comment by lrineau** [comment]
Will be tested tonight in CGAL-4.9-Ic-29, with 12 other PRs. I have verified that at least the polyhedron demo is not broken.

**Comment by lrineau** [comment]
Well, I had to push the commit https://github.com/CGAL/cgal/pull/1038/commits/64ea03f9ec26ffcb0d651ea10bacfe4abb89f622 for it to compile on my machine. It is pushed here, and in `integration`.

**Comment by janetournois** [comment]
this PR had errors in the testsuite for "old" boost versions. 
It should be fixed now

**Pull request merged**